framework/src/userobject/UserObject.C

```diff
@@ -44,12 +44,16 @@ UserObject::validParams()
   params.addParam<bool>("force_postaux", false, "Forces the UserObject to be executed in POSTAUX");
   params.addParam<bool>(
       "force_preic", false, "Forces the UserObject to be executed in PREIC during initial setup");
+  params.addParam<int>("execution_order_group",
+                       0,
+                       "All user objects in the same execution order group are executed "
```

Comment:
not really true. There's ordering within UOs

Reply:
Yes, but how to explain that without writing a whole page?

Reply:
"Please refer to the user object documentation for ordering of user object within a group"

Reply:
How about "Execution order groups are executed in increasing order (e.g., the lowest number is executed first). Within each group, user objects are sorted via a dependency resolver."

Reply:
I'll do a combination of both suggestions

Reply:
Actually the dependency resolver could be misleading, as this only resolves the on element / boundary / node order, not the full execution + finalize. And there is a hard coded order between different UO types.

Reply:
And maybe add something like "Note that negative group numbers may be used to execute groups before the default (0) group."